packages/modular-scripts/src/rename.ts

```diff
@@ -0,0 +1,107 @@
+import { Project } from 'ts-morph';
+import execa from 'execa';
+import actionPreflightCheck from './utils/actionPreflightCheck';
+import getModularRoot from './utils/getModularRoot';
+import getWorkspaceInfo from './utils/getWorkspaceInfo';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as logger from './utils/logger';
+import { cleanGit, stashChanges } from './utils/gitActions';
+
+process.on('SIGINT', () => {
+  stashChanges();
+  process.exit(1);
+});
+
+// Derive types from array; this is handy to iterate against it later
+const dependencyTypes = [
+  'dependencies',
+  'devDependencies',
+  'peerDependencies',
+  'optionalDependencies',
+] as const;
+
+type DependencyType = typeof dependencyTypes[number];
+
+type DependencyObject = {
+  [key in DependencyType]: Record<string, string>;
+};
+
+interface PackageJson extends Partial<DependencyObject> {
+  name: string;
+}
+
+async function rename(
+  oldPackageName: string,
+  newPackageName: string,
+): Promise<void> {
+  const workspace = await getWorkspaceInfo();
+
+  if (!cleanGit(process.cwd())) {
+    throw new Error(
+      'You have unsaved changes. Please save or stash them before we attempt to rename a package.',
+    );
+  }
+
+  try {
+    logger.debug(`Checking for existence of ${oldPackageName} in workspace.`);
```

Comment:
We should check for the existence and collisions before we ask people to stash their changes.

Reply:
Done